cpp/src/arrow/compute/exec/accumulation_queue.h

```diff
@@ -42,9 +45,11 @@ class AccumulationQueue {
 
   void Concatenate(AccumulationQueue&& that);
   void InsertBatch(ExecBatch batch);
+  void InsertAt(ExecBatch batch, size_t idx);
   int64_t row_count() { return row_count_; }
   size_t batch_count() { return batches_.size(); }
   bool empty() const { return batches_.empty(); }
+  void Resize(size_t size) { batches_.resize(size); }
```

Comment:
What are the semantics / expectations here?  Is this more like `std::vector::reserve` or `std::vector::resize`?  Would be good to add a comment pointing out that default-constructed (empty? invalid?) batches will fill the newly added space.  Is the newly added space usable?

Reply:
Added a comment. The new space is usable, it behaves like `resize`.